pkg/apis/core/validation/validation.go

```diff
@@ -286,6 +286,26 @@ var ValidateClassName = apimachineryvalidation.NameIsDNSSubdomain
 // class name is valid.
 var ValidatePriorityClassName = apimachineryvalidation.NameIsDNSSubdomain
 
+// ValidateSecretReference can be use to check whether provided SecretReference object is valid.
+// Secret name is provided and is valid
+// Secret namespace is provided and is valid
+func ValidateSecretReference(secretRef *core.SecretReference, fldPath *field.Path) field.ErrorList {
```

Comment:
Please add unit tests for this function.

Reply:
[TestValidateSecretReference](https://github.com/kubernetes/kubernetes/pull/103024/files#diff-7ee023b4401073292170227985c8ac98015db0c1980657c277a7cea011c24830R17839) function added